src/WebJobs.Script/Content/Script/functions.js

```diff
@@ -69,6 +70,11 @@ function createFunction(f) {
         inputs.unshift(context);
         delete context._inputs;
 
+        if (context._http) {
+            context.res = res();
```

Comment:
Take a look at the crappy logic we have here: https://github.com/Azure/azure-webjobs-sdk-script/blob/dev/src/WebJobs.Script/Binding/HttpBinding.cs#L60 where we're trying to sniff whether the returned object is a json response object v.s. a user defined object. The flaw with the existing logic is that it hinges on a single 'body' property. However, that is faulty if someone tries to return a custom poco with a body property that ISN'T intended to be an http response.

Can your new type help here?